tdms/tests/unit/test_interp_fns.cpp

```diff
@@ -0,0 +1,292 @@
+# include "catch2/catch_test_macros.hpp"
+# include "interpolation_methods.h"
+
+# include <algorithm>
+# include <cmath>
+# include <iostream>
+
+using namespace std;
+
+/**
+ * @brief In the case when cubic interpolation is to be used, check that all polynomial fields up to cubic order are interpolated exactly (to within machine error)
+ *
+ */
+TEST_CASE("interp: cubic interpolation is exact")
+{
+
+    // equidistant points
+    double x[] = {0., 1., 2., 3.};
+    // test acceptence tolerance. Allow for FLOP imprecision and rounding errors
+    // error should be \approx 4 max(c_i) x^2 __DBL__EPSILON, so order 40 * __DBL__EPSILON__
+    double tol = 4e1 * __DBL_EPSILON__;
+
+    // constant field
+    double c0 = 3.1415;
+    double v1 = c0, v2 = c0, v3 = c0, v4 = c0;
+    double v12 = c0, v23 = c0, v34 = c0;
+
+    CHECK(abs(v12 - interp2(v1, v2, v3, v4)) <= tol);
+    CHECK(abs(v23 - interp1(v1, v2, v3, v4)) <= tol);
+    CHECK(abs(v34 - interp3(v1, v2, v3, v4)) <= tol);
+
+    // linear
+    double c1 = -2.7182818;
+    v1 += c1 * x[0];
+    v2 += c1 * x[1];
+    v3 += c1 * x[2];
+    v4 += c1 * x[3];
+    v12 += c1 * (x[1] + x[0]) / 2.;
+    v23 += c1 * (x[2] + x[1]) / 2.;
+    v34 += c1 * (x[3] + x[2]) / 2.;
+
+    CHECK(abs(v12 - interp2(v1, v2, v3, v4)) <= tol);
+    CHECK(abs(v23 - interp1(v1, v2, v3, v4)) <= tol);
+    CHECK(abs(v34 - interp3(v1, v2, v3, v4)) <= tol);
+
+    // quadratic
+    double c2 = 9.81;
+    v1 += c2 * x[0] * x[0];
+    v2 += c2 * x[1] * x[1];
+    v3 += c2 * x[2] * x[2];
+    v4 += c2 * x[3] * x[3];
+    v12 += c2 * (x[1] + x[0]) * (x[1] + x[0]) / 4.;
+    v23 += c2 * (x[2] + x[1]) * (x[2] + x[1]) / 4.;
+    v34 += c2 * (x[3] + x[2]) * (x[3] + x[2]) / 4.;
+
+    CHECK(abs(v12 - interp2(v1, v2, v3, v4)) <= tol);
+    CHECK(abs(v23 - interp1(v1, v2, v3, v4)) <= tol);
+    CHECK(abs(v34 - interp3(v1, v2, v3, v4)) <= tol);
+
+    // cubic
+    double c3 = 4.2;
+    v1 += c3 * x[0] * x[0] * x[0];
+    v2 += c3 * x[1] * x[1] * x[1];
+    v3 += c3 * x[2] * x[2] * x[2];
+    v4 += c3 * x[3] * x[3] * x[3];
+    v12 += c3 * (x[1] + x[0]) * (x[1] + x[0]) * (x[1] + x[0]) / 8.;
+    v23 += c3 * (x[2] + x[1]) * (x[2] + x[1]) * (x[2] + x[1]) / 8.;
+    v34 += c3 * (x[3] + x[2]) * (x[3] + x[2]) * (x[3] + x[2]) / 8.;
+
+    CHECK(abs(v12 - interp2(v1, v2, v3, v4)) <= tol);
+    CHECK(abs(v23 - interp1(v1, v2, v3, v4)) <= tol);
+    CHECK(abs(v34 - interp3(v1, v2, v3, v4)) <= tol);
+}
+
+/**
+ * @brief The hard-coded numerical values for the interpolation constant should all sum to the same value
+ *
+ * Note - the coefficients are not required to sum to unity!
+ */
```

Comment:
Do these comments get rendered in doxygen? If not, do we want them? (maybe not).

Reply:
I don't know what doxygen does regarding these comments - I thought I'd include them for readability reasons nonetheless.

Reply:
If they make it into the doxygen that's a bonus, but if they don't chances are the person running the tests will be a developer with access to the source code anyway. It's also less of a problem since we're making the `TEST_CASE` strings more verbose too.

Reply:
They get ignored. But could be included with adding [tdms/tests/units to the Doxyfile](https://github.com/UCL/TDMS/blob/main/doc/Doxyfile#L22), and adding a 
```c++
/**
 * @file test_interp_fns.cpp
 * Test the blah blah
 */
```
At the top.

... My question was really: In your opinion, is it worth to document the tests in Doxygen? I'm of two minds:

* [No] Because we should stick to the working code in the docs.
* [Yes] Because developers are going to need to understand the tests.

Probably out of scope for this PR.